dhis-2/dhis-support/dhis-support-commons/src/test/java/org/hisp/dhis/commons/config/jackson/JacksonObjectMapperConfigTest.java

```diff
@@ -46,38 +50,44 @@
  */
 public class JacksonObjectMapperConfigTest
 {
-    private ObjectMapper jsonMapper = JacksonObjectMapperConfig.staticJsonMapper();
+    private final ObjectMapper jsonMapper = JacksonObjectMapperConfig.staticJsonMapper();
 
     @Test
     public void testIsoDateSupport()
-        throws JsonProcessingException
     {
-        Map<String, Date> yearTest = jsonMapper.readValue( createDateTest( "2019" ), new DateMapTypeReference() );
-        assertEquals( yearTest.get( "date" ), DateUtils.parseDate( "2019" ) );
-        Map<String, Date> yearMonthTest = jsonMapper.readValue( createDateTest( "2019-01" ),
-            new DateMapTypeReference() );
-        assertEquals( yearMonthTest.get( "date" ), DateUtils.parseDate( "2019-01" ) );
+        assertParsedAsDate( DateUtils.parseDate( "2019" ), "2019" );
```

Comment:
Maybe we can add a few test cases like:
- Complete date
- Date with time
- Date with time and timezone
- Dates in a format that is not ISO

Reply:
Yes, there is an open question as well: we do not accept ISO with 4 digits for nanoseconds part but that is the standard output of ISO DATE TIME format. Feels a little inconsistent.

Reply:
I added more test scenarios.